src/__tests__/operations/property-filter.test.ts

```diff
@@ -587,6 +587,95 @@ describe('extended operators', () => {
+    /* index=1 */ { status: 'ACTIVATING' },
+    /* index=2 */ { status: 'NOT_ACTIVE' },
+    /* index=3 */ { status: 'DEACTIVATING' },
+    /* index=4 */ { status: 'TERMINATED' },
+    /* index=5 */ { status: 0 },
+    /* index=6 */ { status: obj },
+  ];
+
+  function processWithProperty(propertyKey: string, operator: string, token: any, itemsOverride = items) {
+    return processItems(
+      itemsOverride,
+      {
+        propertyFilteringQuery: { operation: 'and', tokens: [{ propertyKey, operator, value: token }] },
+      },
+      {
+        propertyFiltering: {
+          filteringProperties: [
+            {
+              key: 'status',
+              operators: [
+                { operator: '=', tokenType: 'enum' },
+                { operator: '!=', tokenType: 'enum' },
+                { operator: ':', tokenType: 'enum' },
+                { operator: '!:', tokenType: 'value' },
+              ],
+              groupValuesLabel: 'Status values',
+              propertyLabel: 'Status',
+            },
+          ],
+        },
+      }
+    ).items;
+  }
+
+  test.each(['=', '!=', ':'])('matches nothing when token=null and operator="%s"', operator => {
+    const processed = processWithProperty('status', operator, null);
+    expect(processed).toEqual([]);
+  });
+
+  test.each(['=', '!=', ':'])('matches nothing when token="" and operator="%s"', operator => {
+    const processed = processWithProperty('status', operator, '');
+    expect(processed).toEqual([]);
+  });
+
+  test('matches all when token=[] and operator="!="', () => {
+    const processed = processWithProperty('status', '!=', []);
+    expect(processed).toEqual(items);
+  });
+
+  test('matches nothing when token=[] and operator="="', () => {
+    const processed = processWithProperty('status', '=', []);
+    expect(processed).toEqual([]);
+  });
+
+  test.each([{ token: ['NOT_ACTIVE', 'ACTIVE'] }])('matches some when token=$token and operator="="', ({ token }) => {
+    const processed = processWithProperty('status', '=', token);
+    expect(processed).toEqual([items[0], items[2]]);
+  });
+
+  test.each([{ token: [obj, 0] }])('matches some when token=$token and operator="="', ({ token }) => {
+    const processed = processWithProperty('status', '=', token);
+    expect(processed).toEqual([items[5], items[6]]);
+  });
+
+  test.each([{ token: ['ACTIVE', 'NOT_ACTIVE'] }])('matches some when token=$token and operator="!="', ({ token }) => {
+    const processed = processWithProperty('status', '!=', token);
+    expect(processed).toEqual([items[1], items[3], items[4], items[5], items[6]]);
+  });
+
+  test.each([{ token: [0, obj] }])('matches some when token=$token and operator="!="', ({ token }) => {
+    const processed = processWithProperty('status', '!=', token);
+    expect(processed).toEqual([items[0], items[1], items[2], items[3], items[4]]);
+  });
+
+  test.each([['ACTIVE'], 'ACTIVE'])('matches nothing when token=%s and operator=":"', token => {
```

Comment:
why does `:'ACTIVE'` not match anything?

Reply:
Got it, I'd missed the line `{ operator: '!:', tokenType: 'value' }` above, so was missing why `:` and `!:` were behaving differently. Do we need any test coverage for `{ operator: '!:', tokenType: 'enum' }`?

Reply:
I'd say it is enough to ensure the `=` and `!=` are working as expected as every other operator is supposed to return false no matter what is the value.

Btw, we can alternatively throw instead of returning false when an unsupported configuration is used. What do you think?

Reply:
Throwing might be excessive, but I think a dev time warning would make sense, sounds a good improvement.